compiler/rustc_middle/src/dep_graph/mod.rs

```diff
@@ -28,8 +30,12 @@
 //!   could not be instantiated because the current compilation session
 //!   contained no `DefId` for thing that had been removed.
 //!
+//! ## `DepNode` creation and "inference"
+//!
+// Where is the `define_dep_nodes!()` macro?
```

Comment:
The define_dep_nodes macro is defined in dep_node.rs.
It defines the `DepKind` enum, and `DepConstructor`.
`DepNode` is defined in rustc_query_system::dep_graph::dep_node.